olp-cpp-sdk-dataservice-read/src/VersionedLayerClientImpl.cpp

```diff
@@ -517,8 +517,18 @@ bool VersionedLayerClientImpl::RemoveFromCache(
 }
 
 bool VersionedLayerClientImpl::RemoveFromCache(const geo::TileKey& tile) {
-  auto partition_id = tile.ToHereTile();
-  return RemoveFromCache(partition_id);
+  read::QuadTreeIndex cached_tree;
+  if (repository::PartitionsRepository::FindQuadTree(
+          catalog_, settings_, layer_id_, catalog_version_.load(), tile,
+          cached_tree)) {
+    auto data = cached_tree.Find(tile, false);
+    if (!data) {
+      return false;
+    }
```

Comment:
For that method false means some error occurred, e.g. cache is not initialized.
metadata not found in quad tree is 'ok' case, so return true here.